### ISSUE TITLE
Mini Cart block buttons cannot be styled via specific block styles

### ISSUE THREAD

**Issue by nielslange** [woothemes/woocommerce#46106]
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

In https://github.com/woocommerce/woocommerce-blocks/pull/8776 and https://github.com/woocommerce/woocommerce-blocks/pull/8766, we've added support to style the Mini Cart block buttons. However, these buttons can only be styled when opening the Mini Cart template part (within patterns), selecting the corresponding button and then applying the color. The buttons cannot be styled when selecting the corresponding button when opening the styles sidebar, clicking on the `Blocks` link and then searching for the following blocks:
- Mini-Cart View Cart Button
- Mini-Cart Shopping Button
- Mini-Cart Proceed to Checkout Button

Currently, the `block.json` of these blocks has the following definition:

```json
"color": {
	"text": true,
	"background": true
}
```

I thought that changing it as follows would solve the issue.

```json
"color": {
	"text": true,
	"background": true,
	"button": true
}
```

However, this did not work as expected. I did noticed, though, that both the `Mini-Cart View Cart Button` and the `Mini-Cart Proceed to Checkout Button` have a `style` prop, which the `Mini-Cart Shopping Button` does not have. Here are the corresponding `block.tsx` sections:

https://github.com/woocommerce/woocommerce/blob/40be78c265dc80f0a7c2623ab221da04cab1bf05/plugins/woocommerce-blocks/assets/js/blocks/mini-cart/mini-cart-contents/inner-blocks/mini-cart-cart-button-block/block.tsx#L15-L30

https://github.com/woocommerce/woocommerce/blob/40be78c265dc80f0a7c2623ab221da04cab1bf05/plugins/woocommerce-blocks/assets/js/blocks/mini-cart/mini-cart-contents/inner-blocks/mini-cart-checkout-button-block/block.tsx#L19-L35

https://github.com/woocommerce/woocommerce/blob/40be78c265dc80f0a7c2623ab221da04cab1bf05/plugins/woocommerce-blocks/assets/js/blocks/mini-cart/mini-cart-contents/inner-blocks/mini-cart-shopping-button-block/block.tsx#L14-L25

### Expected behavior

I expect that the three buttons, within the Mini Cart block, can be styled via the block specific styles.

### Actual behavior

The three buttons, within the Mini Cart block, cannot be styled via the block specific styles.

### (Semi-)Non-working styles

1. Go to `/wp-admin/site-editor.php`.
2. Click on `Patterns`.
3. Select the `Mini-Cart` template part.
4. Open the `Mini-Cart` template part.
5. Open the `Styles` sidebar (icon in the upper-right corner).
6. In the section `Customize the appearance of specific blocks for the whole site.`, click on the link `Blocks`.
7. Search for the `Mini-Cart View Cart Button`.
8. 🟢 See that the button styles can be applied. 
9. Repeat the previous step with the `Mini-Cart Proceed to Checkout Button`.
10. 🔴 See that no button styles can be applied.
11. Repeat the previous step with the `Mini-Cart Shopping Button`.
12. 🔴 See that no button styles can be applied.

<table>
<tr>
<td valign="top">Mini-Cart View Cart Button
<br><br>
<img width="1352" alt="Screenshot 2024-04-02 at 15 48 39" src="https://github.com/woocommerce/woocommerce/assets/3323310/83d0fc9c-5966-4f57-ac68-3fc8457b2866">
</td>
<td valign="top">Mini-Cart Proceed to Checkout Button
<br><br>
<img width="1352" alt="Screenshot 2024-04-02 at 15 48 22" src="https://github.com/woocommerce/woocommerce/assets/3323310/65af99f5-f234-4e64-acff-6575a6056e4a">

</td>
<td valign="top">Mini-Cart Shopping Button
<br><br>
<img width="1352" alt="Screenshot 2024-04-02 at 15 49 03" src="https://github.com/woocommerce/woocommerce/assets/3323310/04daeaae-a59e-4e44-aea2-47257efeab51">
</td>
</tr>
</table>

### (Semi-)Working styles

1. Go to `/wp-admin/site-editor.php`.
2. Click on `Patterns`.
3. Select the `Mini-Cart` template part.
4. Open the `Mini-Cart` template part.
5. Open the list view  (icon in the upper-left corner). 
6. Open `Mini-Cart View Cart Button` (within `Mini-Cart Contents` → `Filled Mini-Cart view` → `Mini-Cart Footer`).
7. Open the `Styles` sidebar (icon in the upper-right corner).
8. Click on `Colors`.
9. 🟢 See that button styles can be applied.
13. Repeat the previous step with the `Mini-Cart Proceed to Checkout Button` (within `Mini-Cart Contents` → `Filled Mini-Cart view` → `Mini-Cart Footer`).
14. 🟢 See that button styles can be applied.
15. Repeat the previous step with the `Mini-Cart Shopping Button` (within `Mini-Cart Contents` → `Empty Mini-Cart view`).
16. 🔴 See that button styles are not applied correctly in this case.

<table>
<tr>
<td valign="top">Mini-Cart View Cart Button
<br><br>
<img width="1352" alt="Screenshot 2024-04-02 at 15 38 29" src="https://github.com/woocommerce/woocommerce/assets/3323310/2060afb4-c3aa-44a3-8ad4-069708062014">
</td>
<td valign="top">Mini-Cart Proceed to Checkout Button
<br><br>
<img width="1352" alt="Screenshot 2024-04-02 at 15 38 56" src="https://github.com/woocommerce/woocommerce/assets/3323310/35fb93d3-1461-4263-8e1f-a6dcfe72205f">
</td>
<td valign="top">Mini-Cart Shopping Button
<br><br>
<img width="1352" alt="Screenshot 2024-04-02 at 15 42 53" src="https://github.com/woocommerce/woocommerce/assets/3323310/993daf98-5c84-42ce-bda6-c90cca791da7">
</td>
</tr>
</table>

### WordPress Environment

n/a

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woo.com/products/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

**Comment by ralucaStan** [comment]
according to https://github.com/woocommerce/woocommerce/issues/42369, we should only see the Mini Cart block there and not the inner blocks. I'm inclined to close this issue;

@Aljullu I'd really appreciate your input, as you have more context. It's not common to show the inner blocks, right. You mention other blocks impacted by this. Did you find workarounds for them?

**Comment by nielslange** [comment]
> according to https://github.com/woocommerce/woocommerce/issues/42369, we should only see the Mini Cart block there and not the inner blocks. I'm inclined to close this issue;

Inner blocks aside, my intension of creating this issue was to enable merchants to fully control the styles of the Mini Cart block elements. Instead of displaying the inner blocks, we might be able to add the inner block attributes to the parent block, so that they can be controled from there.

**Comment by Aljullu** [comment]
> according to https://github.com/woocommerce/woocommerce/issues/42369, we should only see the Mini Cart block there and not the inner blocks. I'm inclined to close this issue;

Yeah, I also lean towards hiding them from the global styles UI, but not a strong opinion.

> Inner blocks aside, my intension of creating this issue was to enable merchants to fully control the styles of the Mini Cart block elements. Instead of displaying the inner blocks, we might be able to add the inner block attributes to the parent block, so that they can be controled from there.

Users can style them directly when clicking on the block:

![Image](https://github.com/user-attachments/assets/d48f6451-32c9-4b39-913b-b24870002a19)

You mean that's not enough. As mentioned, I don't have a strong opinion, so I'm happy with whatever you folks decide. 🙂 

---

Sidenote: I noticed that changing the background color of the _Mini-Cart Shopping Button_ doesn't work properly, it applies the background outside of the button. Would you like me to create an issue for that?

![Image](https://github.com/user-attachments/assets/e1cc1217-de2e-4949-bf10-39259298aa8d)

**Comment by ralucaStan** [comment]
> Yeah, I also lean towards hiding them from the global styles UI, but not a strong opinion.

I think this path to go and edit buttons out of context is not something common. We didn't have other requests, so I will close the issue for now. 

> Would you like me to create an issue for that?

@Aljullu that would be great! thank you so much for your help

**Comment by Aljullu** [comment]
> @Aljullu that would be great! thank you so much for your help

Done: https://github.com/woocommerce/woocommerce/issues/56074.
